src/main/kotlin/com/devfloor/untitled/article/domain/Article.kt

```diff
@@ -48,14 +48,14 @@ class Article(
     @JoinColumn(name = "author_id")
     val author: User = author
 
-    fun sliceContentByLength(length: Long): String {
-        return if (content.length > length)
+    fun sliceContentByLength(length: Long): String =
+        if (content.length > length)
             content.substring(startIndex = 0, endIndex = (length + 1).toInt())
         else content
-    }
 
-    fun modify(title: String?, content: String) {
-        this.title = title
-        this.content = content
-    }
+    fun modify(title: String?, content: String): Article =
+        this.apply {
+            this.title = title
+            this.content = content
+        }
```

Comment:
이건 굳이 이렇게 쓰는 이유가 있어? 개인적으로 전에 코드가 더 간결하고 이해하기 좋은 것 같아서

Reply:
아 일단 기존코드는 article을 반환하지 않아 ! 아니면 이렇게 쓸수도 있을거같은데, 난 개인적으로 저렇게 짠게 더 간결하고 좋아보여서?
```suggestion
    fun modify(title: String?, content: String): Article {
        this.title = title
        this.content = content
        return this
    }
```

Reply:
아 이거 객체를 반환하는게 이상한 이유가 객체에 대해 내부 값을 변경하는건데 다시 this를 반환하는게 이상한거 같아. 

data class처럼 불변일 경우 재생성하고 반환하는 로직을 가져가는데 이건 Unit 타입으로 프로퍼티만 변경하는게 맞지 않을까? 그렇게 사용할려고 필드도 var로 사용하고 있는거였잖아.

Reply:
아 그러면, 혹시 
Service에서 Article.kt에서 변경한 엔티티를 facade에 넘겨주는게 아니라,
Service의 modify에서 showById 해서 얻어낸 Article을 그대로 facade에 넘겨주자는 말인거야?

Reply:
아 엉엉 그말이었어 ㅋㅋㅋ

```kotlin
fun modify(articleId: Long, request: ArticleModifyRequest): Article {
    return repository.findByIdOrNull(articleId)
        ?.also { it.modify(request.title, request.content) }
        ?: throw IllegalArgumentException()
}
```

이렇게 하면 수정한 게시글을 그대로 파사드에 넘겨줄 수 있으니가?

아 추가적으로 저거 `apply` 쓰는 것도 괜찮은듯?

```kotlin
fun modify(articleId: Long, request: ArticleModifyRequest): Article {
    return repository.findByIdOrNull(articleId)
        ?.apply { modify(request.title, request.content) }
        ?: throw IllegalArgumentException()
}
```